ext/app/electron/config.ts

```diff
@@ -0,0 +1,206 @@
+import * as path from 'path'
+import * as fse from 'fs-extra'
+import * as electron from 'electron';
+import * as ini from 'ini'
+import * as os from 'os'
+import * as log from 'app/server/lib/log';
+import * as packageJson from 'desktop.package.json';
+import {commonUrls} from 'app/common/gristUrls';
+
+const CONFIG_DIR = path.join(electron.app.getPath("appData"), packageJson.name)
+
+const APPDATA_DIR = (process.platform == "win32") ? electron.app.getPath("userData"):
+  path.join(electron.app.getPath("home"), ".local", "share", packageJson.name)
+
+// Electron's app.getPath("userData") uses productName instead of name
+// but productName should be "full capitalized name", not ideal for naming our config directory.
+const DEFAULT_CONFIG_FILE = path.join(CONFIG_DIR, "config.ini")
+const NO_VALIDATION = () => true
+
+/**
+ * Suggest a value for an environment variable. If the variable is already set, do nothing. Otherwise set the value.
+ * @param name The name of the environment variable.
+ * @param value The value to suggest.
+ */
+function suggestEnv(name: string, value: string): void {
+  if (process.env[name] === undefined) {
+    process.env[name] = value;
+  }
+}
+
+type IConfig = {[key: string]: any}
+
+class Config implements IConfig {
+
+  [key: string]: any
+
+  constructor(c: IConfig) {
+    Object.assign(this, c)
```

Comment:
Don't know if I love this, it is a bit too clever? Is there an advantage to having the class with an apply() method itself implement IConfig? Could it just store c and work with that?

Reply:
eslint is not happy with this curious pattern, changed to storing c.